File: src/main/scala/bad/robot/radiate/ui/EmptyTrailingTiles.scala
package bad.robot.radiate.ui

import math._

object EmptyTrailingTiles {
  def tiles(numberOfTiles: Int): EmptyTrailingTiles = new EmptyTrailingTiles(numberOfTiles)
}

class EmptyTrailingTiles(numberOfTiles: Int) extends Tiles {

  val rows = floor(sqrt(numberOfTiles)).toInt

  val columns = rows + (if (numberOfTiles == 1) 0 else 1)

  override def toString = s"$numberOfTiles tiles ($rows, $columns)"
}

File: src/main/java/bad/robot/radiate/ui/EmptyTrailingTiles.java
package bad.robot.radiate.ui;

import static java.lang.String.format;

@Deprecated
class EmptyTrailingTiles implements Tiles {

    private final int numberOfTiles;

    public static EmptyTrailingTiles tiles(int numberOfTiles) {
        return new EmptyTrailingTiles(numberOfTiles);
    }

    EmptyTrailingTiles(int numberOfTiles) {
        this.numberOfTiles = numberOfTiles;
    }

    @Override
    public int rows() {
        return (int) Math.floor(Math.sqrt(numberOfTiles));
    }

    @Override
    public int columns() {
        int rows = rows();
        return rows + (numberOfTiles == 1 ? 0 : 1);
    }

    @Override
    public String toString() {
        return format("%d tiles (%d, %d)", numberOfTiles, rows(), columns());
    }
}
